src/context.rs

```diff
@@ -1789,19 +1836,43 @@ impl Context {
             Error::local_error(ErrorKind::MissingAuthSession)
         })
     }
+
+    /// Function that returns the handle as an ESYS_TR if it
+    /// was valid or an error if it was None.
+    fn ensure_not_none(handle: &ObjectHandle, fn_name: &str) -> Result<ESYS_TR> {
+        if *handle != ObjectHandle::None {
+            Ok((*handle).into())
+        } else {
+            error!("ObjectHandle::None cannot be used in call to {}", fn_name);
+            Err(Error::local_error(ErrorKind::InvalidParam))
+        }
+    }
```

Comment:
No it got placed there because I was doing stuff in Context and needed the function. But you are right it is not really tied to the Context struct so it should probably not be there.